apps/server/src/shared/domain/service/permission.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@nestjs/common';
+import { Role, User } from '../entity';
+
+@Injectable()
+export class PermissionService {
+	/**
+	 * Recursively resolve all roles and permissions of a user.
+	 * IMPORTANT: The role collections of the user and nested roles will not be loaded lazily.
+	 * Please make sure you populate them before calling this method.
+	 * @param user
+	 * @returns
+	 */
+	resolveRolesAndPermissions(user: User): [Role[], string[]] {
+		const rolesAndPermissions = this.resolveRolesAndPermissionsByRoles(user.roles.getItems());
```

Comment:
let speak about getItems in line 14 and 25

Reply:
yes. same issue as above. we should clarify how to pass entity lists as parameters.